devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,550 @@
+					break
+				case "auto":
+					getThermostatFanModeCommands(CUSTOM_FAN_MODE_AUTO)
+					break
+				case "circulate":
+					getThermostatFanModeCommands(CUSTOM_FAN_MODE_CIRCULATE)
+					break
+			}
+		}
+	} else {
+		log.debug "Unsupported fan mode $mode"
+	}
+}
+
+def fanOn() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_ON)
+}
+
+def fanAuto() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_AUTO)
+}
+
+def fanCirculate() {
+	getThermostatFanModeCommands(CUSTOM_FAN_MODE_CIRCULATE)
+}
+
+def getThermostatFanModeCommands(mode) {
+	if (mode) {
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE, DataType.ENUM8, mode),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_MODE)
+		], 500)
+	}
+}
+
+def setFanSpeed(speed) {
+	log.debug "setFanSpeed: ${speed}"
+
+	if (speed == 0 || speed >= 4) { //if by any chance user selects 0 or a value higher than 3, it fan will be set to AUTO
+		speed = 3
+	} else {
+		speed = speed - 1
+	}
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_SPEED, DataType.ENUM8, speed),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_FAN_SPEED),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def setThermostatMode(mode) {
+	log.debug "set mode $mode (supported ${state.supportedThermostatModes})"
+	if (state.supportedThermostatModes?.contains(mode)) {
+		switch (mode) {
+			case "auto":
+				auto()
+				break
+			case "cool":
+				cool()
+				break
+			case "heat":
+				heat()
+				break
+			case "off":
+				off()
+				break
+		}
+	} else {
+		log.debug "Unsupported mode $mode"
+	}
+}
+
+def off() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_OFF),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def auto() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_AUTO),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def cool() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_COOL),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def heat() {
+	delayBetween([
+			zigbee.writeAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE, DataType.ENUM8, THERMOSTAT_MODE_HEAT),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_MODE),
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, CUSTOM_THERMOSTAT_OPERATING_STATE)
+	], 500)
+}
+
+def ping() {
+	log.debug "ping"
+	refresh()
```

Comment:
the ping method should not be querying all of these attributes. Just temperature is fine.

Reply:
fixed